openedx/core/djangoapps/xblock/runtime/blockstore_runtime.py

```diff
@@ -148,13 +153,45 @@ def save_block(self, block):
         olx_path = definition_key.olx_path
         blockstore_api.write_draft_file(draft_uuid, olx_path, olx_str)
         # And the other files, if any:
+        olx_static_path = os.path.dirname(olx_path) + '/static/'
         for fh in static_files:
-            raise NotImplementedError(
-                "Need to write static file {} to blockstore but that's not yet implemented yet".format(fh.name)
-            )
+            new_path = olx_static_path + fh.name
+            blockstore_api.write_draft_file(draft_uuid, new_path, fh.data)
         # Now invalidate the blockstore data cache for the bundle:
         BundleCache(definition_key.bundle_uuid, draft_name=definition_key.draft_name).clear()
 
+    def _lookup_asset_url(self, block, asset_path):
+        """
+        Return an absolute URL for the specified static asset file that may
+        belong to this XBlock.
+
+        e.g. if the XBlock settings have a field value like "/static/foo.png"
+        then this method will be called with asset_path="foo.png" and should
+        return a URL like https://cdn.none/xblock/f843u89789/static/foo.png
+
+        If the asset file is not recognized, return None
+        """
+        if '..' in asset_path:
+            return None  # Illegal path
```

Comment:
Nit: We're doing enough of these checks that it might be worth extracting into something more centralized for valid path checking.

Reply:
Yeah I probably didn't even need to put this here given the other checks, but I figured it didn't hurt.